### PR TITLE
Create Budget App

### PR DESCRIPTION
**Pull request by LIBERCOSOFT**
# Create a Budget App with rails
## In this PR;
- Splash screen
    - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign-up and login pages
    - The user should be able to register in the app with full name, email and password (all mandatory).
    - The user can log into the app using email and password.
    - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (categories page)
    - When the user logs in, they are presented with the categories page.
    - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
    - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
    - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
    - For a given category, the list of transactions is presented, ordered by the most recent.
    - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
    - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
    - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page
    - The user fills out a form to create a new transaction with:
        - name (mandatory)
        - amount (mandatory)
        - categories (mandatory at least one)
    - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
    - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Create unit and integration tests for all the most important components of your RoR application.
- Link to video: https://www.loom.com/share/63fd881693424bd08ec9d32735f82ee1

### CODE REVIEW

File: app/views/groups/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <div class="my-auto mx-auto">
+    <h2 class="text-2xl"> <%= link_to group.name, group_path(group.id) %> </h2>
+    <p class="mt-4"><%= group.created_at.strftime("%d %b %Y") %></p>
+    </div>
+    <p class="my-auto mx-auto">$<%= total_price %></p>
+  </div>
+<% end %>
+
+<div class="w-1/2 text-center my-8 mx-auto">
+  <%= link_to 'Add a new category', new_group_path, class: "bg-secondary text-white text-2xl p-4" %>
```
Comment (rustam-tolipov) [inline app/views/groups/index.html.erb]: **You did grate job! I like the implementations of tests!**

However, we have some places that need to be fixed. 

- [ ] Your `Add a new category` button is not styling correctly. Could you fix the issue and make sure it is placed as expected? It should be at the bottom as well. Also, add transaction button has a similar problem. Kindly fix them as well.

Screenshot 2023-02-17 at 16 33 15

File: Gemfile [hunk 2]
```diff
@@ -1,31 +1,31 @@
-source "https://rubygems.org"
+source 'https://rubygems.org'
```
Comment (rustam-tolipov) [inline Gemfile:1]: Let's speak about your README. 

- [x] 1. Your project should have a deployment link in the readme. Please, add the deployment link in the readme, also make sure the app is working as expected.

- [x] 2. I think you should give proper credit to the designer. Adding only a link to the design is not enough. Thank you!

File: spec/requests/groups_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  #   end
+
+  #   it 'Should render the correct template' do
+  #     expect(response).to render_template(:index)
+  #   end
+
+  #   it 'should render the correct text in the template' do
+  #     expect(response.body).to include 'All Categories'
+  #   end
+  # end
```
Comment (rustam-tolipov) [inline spec/requests/groups_spec.rb]: - [x] Kindly remove commented codes from your file!

File: test/models/budget_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class BudgetTest < ActiveSupport::TestCase
```
Comment (rustam-tolipov) [inline test/models/budget_test.rb]: - [x] Kindly remove commented codes from your file!

File: test/models/group_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class GroupTest < ActiveSupport::TestCase
```
Comment (rustam-tolipov) [inline test/models/group_test.rb]: - [x] Kindly remove commented codes from your file!

File: test/models/user_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
```
Comment (rustam-tolipov) [inline test/models/user_test.rb]: - [x] Kindly remove commented codes from your file!

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @LIBERCOSOFT,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_I like the way you tested with capybara_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 7]
```diff
@@ -50,13 +51,11 @@
...
-<!-- LIVE DEMO -->
+## 🚀 Live Demo <a name="live-demo"></a>
 
-<!-- ## 🚀 Live Demo <a name="live-demo"></a>
+[Live Demo Link](https://budget-app-9ny9.onrender.com/)
 
-- [Live Demo Link](https://yourdeployedapplicationlink.com)
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p> -->
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (BenMukebo) [inline README.md:58]: - [ ] As requested by the previous reviewer, kindly make sure that:
1. The project is deployed and accessible online, (you can use Heroku, Netlify or GitHub Pages).
2. The project is deployed properly, and the home and detail page works as expected.
3. A link to the deployed project is present in the README.
- Kindly address this 💯 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
## Hi @LIBERCOSOFT, 

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


### To Highlight 🎉
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Good presentation video. ✔️ 

## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/groups/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    <h2 class="text-2xl"> <%= link_to group.name, group_path(group.id) %> </h2>
+    <p class="mt-4"><%= group.created_at.strftime("%d %b %Y") %></p>
+    </div>
+    <p class="my-auto mx-auto">$<%= total_price %></p>
+  </div>
+<% end %>
+
+<div class="w-1/2 text-center my-8 mx-auto">
+  <%= link_to 'Add a new category', new_group_path, class: "bg-secondary text-white text-base p-2 md:text-xl " %>
+</div>
```
Comment (BenMukebo) [inline app/views/groups/index.html.erb]: - Kindly improve the design, Check these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
* Colors
* Typography: font face, size and weight.
* Layout: composition and space between elements.
- [ ] As requested by the previous reviouwer, kindly follow the provided design (width, font, size, height, ...)and make sure you place the  "add a new category" button at the botton of your page.
- [ ] The  transaction button has a similar problem. Kindly fix them as well.

Your Implementation | Original Design
:---:|:---:
image | image

File: README.md (hunk 7 above)
Comment (LIBERCOSOFT) [reply README.md:58]: @BenMukebo, the link to the deployed project is in my readme file

**Review by uwadonat — APPROVED** [review]
## Hi @LIBERCOSOFT,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Professional readme file ✔️
- Well organized and clean project ✔️
- No linter error ✔️
approval1


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**